src/plugins/list-enhanced.js

```diff
@@ -0,0 +1,344 @@
+/**
+ * SCEditor Inline-Code Plugin for BBCode format
+ * http://www.sceditor.com/
+ *
+ * Copyright (C) 2011-2013, Sam Clarke (samclarke.com)
+ *
+ * SCEditor is licensed under the MIT license:
+ *	http://www.opensource.org/licenses/mit-license.php
+ *
+ * @fileoverview SCEditor list enhancement plugin
+ * This plugin expands the implementation of the list commands, adding dropdown
+ * options to select between different list types
+ * @author Alex Betis
+ */
+
+(function (sceditor) {
+	'use strict';
+
+	var dom = sceditor.dom;
+
+	/**
+	* Fixes a bug in FF where it sometimes wraps
+	* new lines in their own list item.
+	* See issue #359
+	*/
+	function fixFirefoxListBug(editor) {
```

Comment:
What prevents you from making an algorithm that works on all browsers?

Reply:
I follow the original list commands implementation:
```
	// START_COMMAND: Bullet List
	bulletlist: {
		exec: function () {
			fixFirefoxListBug(this);
			this.execCommand('insertunorderedlist');
		},
		tooltip: 'Bullet list'
	},
	// END_COMMAND
	// START_COMMAND: Ordered List
	orderedlist: {
		exec: function () {
			fixFirefoxListBug(this);
			this.execCommand('insertorderedlist');
		},
		tooltip: 'Numbered list'
	},
	// END_COMMAND
```

Reply:
Fair enough